src/core/SP_reader.sol

```diff
@@ -0,0 +1,217 @@
+// SPDX-License-Identifier: GNU AGPLv3
+pragma solidity ^0.8.20;
+
+import { OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { Attestation as SPAttestation } from "../models/Attestation.sol"; // Ignoring Offchain Attestations
+import { Attestation as VeraxAttestation, IVerax } from "src/interfaces/IVerax.sol";
+import { SP } from "src/core/SP.sol";
+/*
+    Found Information Above on Github:
+    - Verax's Attestation struct 
+    - Interface I created to call a certain function -- getAttestation()
+*/
+
+contract Reader is OwnableUpgradeable {
```

Comment:
Try to name the contract more descriptively and match the file name, e.g. `SPReader`